StoreKitUnitTests/TrialOrIntroPriceEligibilityCheckerSK2Tests.swift

```diff
@@ -131,4 +133,122 @@ class TrialOrIntroPriceEligibilityCheckerSK2Tests: StoreKitConfigTestCase {
             expect(receivedEligibility.status) == expected[product]
         }
     }
+
+    @available(iOS 15.0, tvOS 15.0, macOS 12.0, watchOS 8.0, *)
+    func testCheckEligibilityForProductIsEligible() async throws {
+        try AvailabilityChecks.iOS15APIAvailableOrSkipTest()
+
+        let productIdentifiers = Set([
+            "com.revenuecat.monthly_4.99.1_week_intro"
+        ])
+
+        let sk2Product = try await ProductsFetcherSK2().products(identifiers: productIdentifiers).first
+        let receivedProduct = try XCTUnwrap(sk2Product)
+        let storeProduct = StoreProduct.from(product: receivedProduct)
+
+        var completionCalled = false
+
+        let status = try await withCheckedThrowingContinuation({ (continuation: ContinuationStatusResult) in
+            self.trialOrIntroPriceEligibilityChecker!.checkEligibility(product: storeProduct) { status in
```

Comment:
[not needed for this PR] perhaps we should just make the `trialOrIntroPriceEligibilityChecker` implicitly unwrapped so we don't have to do the `!` on every test

Reply:
actually, it's already declared as being unwrapped in line 25 so this wasn't necessary, I'll remove from all of the tests